src/psyclone/psyir/nodes/call.py

```diff
@@ -68,44 +69,179 @@ def __init__(self, routine, parent=None):
-        symbol, and a list of child nodes for its arguments.
+        symbol, and a list of child nodes (or name and node tuple) for
+        its arguments.
 
         :param routine: the routine that class cls calls.
         :type routine: py:class:`psyclone.psyir.symbols.RoutineSymbol`
-        :param arguments: the arguments to this routine. These are \
-            added as child nodes.
-        :type arguments: list of :py:class:`psyclone.psyir.nodes.DataNode`
+        :param arguments: the arguments to this routine, and/or \
+            2-tuples containing an argument name and the \
+            argument. Arguments are added as child nodes.
+        :type arguments: List[ \
+            Union[:py:class:``psyclone.psyir.nodes.DataNode``, \
+                  Tuple[str, :py:class:``psyclone.psyir.nodes.DataNode``]]]
+
+        :returns: an instance of cls.
+        :rtype: :py:class:`psyclone.psyir.nodes.Call` or a subclass thereof.
 
         :raises GenerationError: if the routine argument is not a \
             RoutineSymbol.
         :raises GenerationError: if the arguments argument is not a \
             list.
-
-        :returns: an instance of cls.
-        :rtype: :py:class:`psyclone.psyir.nodes.Call` or a subclass thereof.
+        :raises GenerationError: if the contents of the arguments \
+            argument are not the expected type.
 
         '''
         if not isinstance(routine, RoutineSymbol):
             raise GenerationError(
-                "Call create routine argument should be a RoutineSymbol but "
-                "found '{0}'.".format(type(routine).__name__))
+                f"Call create routine argument should be a RoutineSymbol but "
+                f"found '{type(routine).__name__}'.")
         if not isinstance(arguments, list):
             raise GenerationError(
-                "Call create arguments argument should be a list but found "
-                "'{0}'.".format(type(arguments).__name__))
+                f"Call create arguments argument should be a list but found "
+                f"'{type(arguments).__name__}'.")
 
         call = cls(routine)
-        call.children = arguments
+        for arg in arguments:
+            name = None
+            if isinstance(arg, tuple):
+                if not len(arg) == 2:
+                    raise GenerationError(
+                        f"If a child of the children argument in create "
+                        f"method of Call class is a tuple, it's "
+                        f"length should be 2, but found {len(arg)}.")
+                if not isinstance(arg[0], str):
+                    raise GenerationError(
+                        f"If a child of the children argument in create "
+                        f"method of Call class is a tuple, its first "
+                        f"argument should be a str, but found "
+                        f"{type(arg[0]).__name__}.")
+                Call._validate_name(arg[0])
```

Comment:
I think we don't need this validate call since is already called inside append_named_arg below

Reply:
Oh, yes.